source/extensions/filters/http/transformation/inja_transformer.cc

```diff
@@ -394,6 +400,28 @@ json TransformerInstance::extracted_callback(const inja::Arguments &args) const
   return "";
 }
 
+json TransformerInstance::data_source_callback(const inja::Arguments &args) const {
+  const auto& ctx = tls_.getTyped<ThreadLocalTransformerContext>();
+  const std::string &name = args.at(0)->get_ref<const std::string &>();
+  const auto value_it = ctx.data_sources_->find(name);
+  if (value_it != ctx.data_sources_->end()) {
+    return value_it->second->data();
+  }
+  return "";
```

Comment:
Should we make this with a default return?

Reply:
this return represents a user error. i.e. the user asking for a data source the user did not define. so i don't think we need a default here, as this should never happen.